test/core/annotations/test_before_after_test_annotation.pkb

```diff
@@ -0,0 +1,138 @@
+            procedure beforetest_one_err_procedure;
+            
+            procedure test2;
+        end;
+    ';
+
+    l_test_package_body := '
+        create or replace package body dummy_before_after_test is
+            procedure clean_global_variables is
+            begin
+              l_dummy_1 := null;
+              l_dummy_2 := null;
+              l_dummy_3 := null;  
+            end;
+        
+            procedure beforetest_one_ext_procedure is
+            begin
+              ut3.ut.expect(l_dummy_1).to_equal(1);
+            end;
+            
+            procedure beforetest_multi_ext_procedure
+            is
+            begin
+              ut3.ut.expect(l_dummy_1).to_equal(1);  
+              ut3.ut.expect(l_dummy_2).to_equal(2);
+              ut3.ut.expect(l_dummy_3).to_equal(3);
+            end;
+            
+            procedure beforetest_one_err_procedure
+            is
+            begin
+              null;
+            end;
+            
+            procedure test2 is
+            begin
+              l_dummy_2 := 2;  
+            end;
+        end;
+    ';
+    
+    l_dummy_utility_pkg_spec := '
+      create or replace package dummy_utility_pkg_1 is
+        procedure test1;
+        
+        procedure test3;
+      end;
+    ';
+    
+    l_dummy_utility_pkg_body := '
+      create or replace package body dummy_utility_pkg_1 is
+        procedure test1 is
+        begin
+          dummy_before_after_test.l_dummy_1 := 1;
+        end;
+        
+        procedure test3 is
+        begin
+          dummy_before_after_test.l_dummy_3 := 3;
+        end;
+      end;      
+    ';
+    
+    execute immediate l_test_package_spec;
+    execute immediate l_test_package_body;
+    execute immediate l_dummy_utility_pkg_spec;
+    execute immediate l_dummy_utility_pkg_body;
+
+    --Execute the tests and recolect the results
+    select * bulk collect into l_test_results from table(ut3.ut.run(('dummy_before_after_test')));
+    
+    drop_package('dummy_utility_pkg_1');
+    drop_package('dummy_before_after_test');
+
+    g_tests_results := ut3.ut_utils.table_to_clob(l_test_results);
+  end;
+  
+  procedure beforetest_one_ext_procedure
+  is
+  begin
+    ut.expect(g_tests_results).to_match('^\s*Beforetest with call to procedure external to the test package \[[\.0-9]+ sec\]\s*$','m');
+    ut.expect(g_tests_results).not_to_match('beforetest_one_ext_procedure');
+  end;
+  
+  procedure beforetest_multi_ext_procedure
+  is
+  begin
+    ut.expect(g_tests_results).to_match('^\s*Beforetest with call to multi procedures external and interal to the test package \[[\.0-9]+ sec\]\s*$','m');
+    ut.expect(g_tests_results).not_to_match('beforetest_multi_ext_procedure');
+  end;
+  
+  procedure beforetest_one_err_procedure
+  is
+  begin
+    ut.expect(g_tests_results).not_to_match('^\s*Beforetest with call to multi procedures where one does not exist \[[\.0-9]+ sec\]\s*$','m');
```

Comment:
Can you make expectation so that it says what `is` expected rather than what `is not` expected?
I *guess*, we expect a failing test here, but the line isn't explicit about that.

Reply:
What will happen with the third `beforetest` procedure when the second one cannot be found?
Will it be executed or not?
It would be great to be able to check that.
Maybe you could have a global boolean variable in `dummy_utility_pkg_1` for each procedure to check if procedure was executed and then have expectation on that?
```sql
  v_test1 boolean;
  v_test2 boolean;
  v_test3 boolean;
  v_beforetest_one_ext boolean;
  v_beforetest_multi_ext boolean;
  v_beforetest_one_err boolean;
```

Reply:
Actually, when a beforetest is not found it does not execute the others procedures either the test. For example:
```
  --%beforetest(dummy_utility_pkg_1.test4, non_existent_procedure, ut3_tester.dummy_utility_pkg_1.test4)
  procedure beforetest_one_err_procedure;
```
It only will execute dummy_utility_pkg_1.test4 nothing more. is is the behaviour we want to?